extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java

```diff
@@ -52,12 +56,16 @@
     }
 
     public Handler<RoutingContext> uiHandler(String graphqlUiFinalDestination,
-            String graphqlUiPath, SmallRyeGraphQLRuntimeConfig runtimeConfig) {
+            String graphqlUiPath, List<FileSystemStaticHandler.StaticWebRootConfiguration> webRootConfigurations,
+            SmallRyeGraphQLRuntimeConfig runtimeConfig, ShutdownContext shutdownContext) {
 
         if (runtimeConfig.enable) {
-            return new SmallRyeGraphQLStaticHandler(graphqlUiFinalDestination, graphqlUiPath);
+            WebJarStaticHandler handler = new WebJarStaticHandler(graphqlUiFinalDestination, graphqlUiPath,
+                    webRootConfigurations);
+            shutdownContext.addShutdownTask(new ShutdownContext.CloseRunnable(handler));
```

Comment:
@mkouba @cescoffier I was going back to this patch and was wondering why we needed to do that? Shouldn't the handler be closed by Vert.x itself?

Reply:
https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/FileSystemStaticHandler.java#L166

This is the handler in question which should be closed.
At first, I also did hopped that closeable are automaticly closed. I set a breakpoint in the close method though. Never got invoked.

This is why I decided to add the shutdownhandlers.

Is there any better / automatic way with Vert.x?

Reply:
Ah OK, I missed you specifically implemented `Closeable`. Maybe @mkouba will have some ideas about how we could improve it and close `Closeable` handlers automatically.
If not, that's not a biggy.

I was going through this patch again to move Quarkus GitHub App to this new pattern.